src/comm/aca_message_pulsar_consumer.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright 2019 The Alcor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "aca_message_pulsar_consumer.h"
+#include "goalstate.pb.h"
+#include "aca_comm_mgr.h"
+#include "aca_log.h"
+
+using aca_comm_manager::Aca_Comm_Manager;
+using pulsar::Client;
+using pulsar::ClientConfiguration;
+using pulsar::ConsumerConfiguration;
+using pulsar::Consumer;
+using pulsar::Message;
+using pulsar::Result;
+
+
+namespace messagemanager
+{
+MessagePulsarConsumer::MessagePulsarConsumer(string brokers, string subscription_name)
+{
+  setBrokers(brokers);
+  setSubscriptionName(subscription_name);
+
+  // TO DO: Construct the configuration
+
+  ACA_LOG_DEBUG("Broker list: %s\n", this->brokers_list.c_str());
+  ACA_LOG_DEBUG("Consumer subscription name: %s\n", this->subscription_name.c_str());
+
+  // Create the consumer and client
+  this->ptr_consumer = new Consumer();
+  this->ptr_client= new Client(brokers,this->client_config);
+}
+
+MessagePulsarConsumer::~MessagePulsarConsumer()
+{
+  //delete ptr_consumer;
+  delete ptr_client;
+}
+
+string MessagePulsarConsumer::getBrokers() const
+{
+  return this->brokers_list;
+}
+
+string MessagePulsarConsumer::getLastTopicName() const
+{
+  return this->topic_name;
+}
+
+string MessagePulsarConsumer::getSubscriptionName() const
+{
+  return this->subscription_name;
+}
+
+void MessagePulsarConsumer::setSubscriptionName(string subscription_name)
+{
+  this->subscription_name = subscription_name;
+}
+
+bool MessagePulsarConsumer::consumeDispatched(string topic)
+{
+  alcor::schema::GoalState deserialized_GoalState;
+  alcor::schema::GoalStateOperationReply gsOperationalReply;
+  int rc;
+  int overall_rc = EXIT_SUCCESS;
+	Result result;
```

Comment:
please use spaces instead of tabs for indentation. Many IDE has a feature called "untabify" which can convert the whole file for you.